### DISCUSSION TITLE
Missing Table of Contents and Shortcut Features

### DISCUSSION THREAD

**Discussion by Atom-U** [chrede88/qubt#154] [Ideas]
Hi,

I'm currently using this theme to create **long-form blog posts**, and while I really appreciate its design and functionality, but the lack of a Table of Contents (TOC) is a dealbreaker. 
Navigating through long posts without this feature makes it hard to stay organized, and as a result, I’ll be **switching** to another theme that offers better support for content navigation.

It would be great if you could add this feature in the future, as it’s essential for long articles.

Thanks 
Atom

**Comment by chrede88** [comment]
Thank you for the feedback :slightly_smiling_face: I'll try to find the time to add this. I'll post here if/when I get to it.

Do you mind linking to a post that you would like to have a TOC on?
Do you have a preference for how many levels the TOC should display?

**Comment by chrede88** [comment]
I've implemented a simple TOC. Let me know what you think. If you like it, I'll push a new release.
The TOC will display headings from level h2 to h4 (`##` to `####`).
<img width="366" alt="Screenshot 2024-10-20 at 8 08 24 PM" src="https://github.com/user-attachments/assets/19614dac-1de6-4330-9f0d-b4302afc5fd7">

**Reply by Atom-U** [reply]
thanks for your reactivity

**Comment by chrede88** [comment]
Added to release [v1.4.0](https://github.com/chrede88/qubt/releases/tag/v1.4.0).
